upload/views/commits.py

```diff
@@ -23,12 +26,41 @@ class CommitViews(ListCreateAPIView, GetterMixin):
         GlobalTokenAuthentication,
         OrgLevelTokenAuthentication,
         RepositoryLegacyTokenAuthentication,
+        TokenlessAuthentication,
     ]
 
     def get_queryset(self):
         repository = self.get_repo()
         return Commit.objects.filter(repository=repository)
 
+    def _possibly_fix_branch_name(self, request) -> None:
+        """Avoids users being able to overwrite coverage info for a branch
+        that exists in the upstream repo with coverage for their fork branch.
+        By pre-pending the fork name to the branch
+        (the CLI might have done this already)
+        """
+        # This only affects tokenless uploads
+        if not isinstance(request.auth, TokenlessAuth):
+            return
+        # Notice that at this point we already validated that this fork_slug
+        # is the correct repo from the head of a PR to the upstream repo
+        # with the git provider
+        fork_slug = request.headers.get("X-Tokenless", None)
+        branch_info = request.data.get("branch")
+        if branch_info is None:
+            # There should always be a branch in the request
+            raise ValidationError("missing branch")
+        # The CLI might have pre-prended the branch with something already
+        if ":" in branch_info:
+            _, branch_info = branch_info.split(":")
```

Comment:
Would branch_info always have 2 items if it had a ":"?

Reply:
I think so. we do `fork_slug:branch`. And as mentioned by Dana [here](https://github.com/codecov/codecov-cli/pull/331#discussion_r1392259208) branches can't have `:` in them. I don't think repos can either.